### PR TITLE
Add Windows Authentication auth option for SQL Server and dynamic switch to msnodesqlv8 if chosen

### PR DESCRIPTION
**Pull request by mattpcanales**
With Microsoft ending support for Azure Data Studio in Feb 2026, I am left without a clean, DB focused tool that I can use to keep my SQL queries separate from VSCode at work. Beekeeper Studio seems like a great fit, but the requirement of storing credentials isn't ideal for my corporate windows environment, and we aren't fully on Entra ID yet. Unfortunately the pure javascript Tedious driver cannot leverage Microsoft's SSPI bindings required for kerberos without a wrapper of some kind.

There are several open issues on beekeeper-studio that are requesting a solution to this:
#1186, #48, and #534 
as well as more on tedious github repo to show discussion on tedious limitations/roadmap.
[#660](https://github.com/tediousjs/tedious/issues/660), [#415](https://github.com/tediousjs/tedious/issues/415), [#1009](https://github.com/tediousjs/tedious/issues/1009), and [#306](https://github.com/tediousjs/tedious/issues/306)

So I know I am not alone in hoping for this to be addressed, and with ADS end of support I assume this is only increasing recently

The cleanest solution I could come up with involved an added optional dependency for msnodesqlv8 which does offer this support. On top of this node module, the SQL driver on the local system must be ODBC 17 or 18 to fully support TLS1.2
The basic SQL Server driver works without TLS 1.2, but is no longer officially supported by Microsoft past SQL 2012.

I do not notice any difference in the experience between connecting with integrated auth with msnodesqlv8 and using password with mssql + the tedious driver, but I haven't tested every edge case.

I ran the unit test in /app/studio and all passed.

If you have any feedback, even harsh criticism please share. I am happy to consider this just a learning experience since it is my first work with electron and first time contributing to a project. I would love to learn if there is a plan to include this feature some other way, and what architecture is planned to do so, or if the plan is to simply support Azure auth only. 

Thank you for all your work!

### CODE REVIEW

**Comment by rathboma** [comment]
Hey! Wow, this is super cool. Thanks for the contribution. We'll review the code and get back to you.

Off the top of my head I bet there are a couple of changes needed in esbuild.js to externalize the dependency, and might not work when using the full 'build' (vs in dev mode) without that change.

I also need to think about how we properly integration test this. Would be good to have some way to automated-test this with a windows VM / container of some sort.

File: apps/studio/src/lib/db/clients/sqlserver.ts [hunk 1]
```diff
@@ -1072,7 +1072,50 @@ export class SQLServerClient extends BasicDatabaseClient<SQLServerResult, Transa
     await super.connect();
 
     this.dbConfig = await this.configDatabase(this.server, this.database, signal)
-    this.pool = await new ConnectionPool(this.dbConfig).connect();
+
+    if (this.server.config.windowsAuthEnabled) {
+      if (process.platform !== 'win32') {
+        throw new Error('Windows Authentication is only supported on Windows.')
+      }
```
Comment (not-night-but) [inline apps/studio/src/lib/db/clients/sqlserver.ts:1079]: We should use platform info here instead.
```suggestion
      import platformInfo from "@/common/platform_info"
      if (!platformInfo.isWindows) {
        throw new Error('Windows Authentication is only supported on Windows.')
      }
```

File: apps/studio/src/lib/db/clients/sqlserver.ts [hunk 2]
```diff
@@ -1072,7 +1072,50 @@ export class SQLServerClient extends BasicDatabaseClient<SQLServerResult, Transa
...
+      // defaulting to "SQL Server Native Client 11.0"), passes
+      // Trusted_Connection as a boolean (should be yes/no), and doesn't include TrustServerCertificate.
+      // beforeConnect also preserves mssql's instance name and SSH tunnel server handling.
+      const trustCert = this.dbConfig.options?.trustServerCertificate
+      this.pool = await new sqlWindows.ConnectionPool({
+        ...this.dbConfig,
+        beforeConnect: (cfg: any) => {
+          cfg.conn_str = cfg.conn_str
+            .replace(/Driver=[^;]*/i, `Driver={${driver}}`)
+            .replace(/Trusted_Connection=[^;]*/i, 'Trusted_Connection=yes')
```
Comment (not-night-but) [inline apps/studio/src/lib/db/clients/sqlserver.ts:1112]: Looks like we can pass the driver if we update to at least [v12.2.0](https://github.com/tediousjs/node-mssql/releases/tag/v12.2.0)
Usage: 
https://github.com/tediousjs/node-mssql#windows-authentication-example-using-msnodesqlv8

It also looks like you would be able to pass trusted connection in the config there as well

File: apps/studio/src/components/connection/SqlServerForm.vue [hunk 3]
```diff
@@ -5,11 +5,12 @@
...
       <select name="" v-model="authType" id="">
         <option value="default">Username / Password</option>
+        <option value="windows">Windows Authentication</option>
         <option :key="`${t.value}-${t.name}`" v-for="t in authTypes" :value="t.value">{{t.name}}
         </option>
       </select>
     </div>
-    <common-server-inputs v-show="!azureAuthEnabled" :config="config">
+    <common-server-inputs v-show="!azureAuthEnabled" :config="config" :hide-credentials="windowsAuthEnabled">
       <div class="advanced-connection-settings">
```
Comment (not-night-but) [inline apps/studio/src/components/connection/SqlServerForm.vue:14]: We should think about cleaning up this form for Windows Auth. Looking at how SSMS handles this, we probably don't need these sql server options here? Correct me if I'm wrong please :)

File: apps/studio/src/components/connection/SqlServerForm.vue [hunk 4]
```diff
@@ -5,11 +5,12 @@
       <!-- need to take the value -->
       <select name="" v-model="authType" id="">
         <option value="default">Username / Password</option>
+        <option value="windows">Windows Authentication</option>
```
Comment (not-night-but) [inline apps/studio/src/components/connection/SqlServerForm.vue:8]: We just shouldn't show this option if we're not on windows. In the frontend you should just be able to use `window.platformInfo.isWindows` to check (`apps/studio/src/common/IPlatformInfo.ts` will show you what all is available on that object)

File: apps/studio/electron-builder-config.js [hunk 5]
```diff
@@ -37,7 +37,8 @@ module.exports = {
   ],
   afterPack: "./build/afterPack.js",
   asarUnpack: [
-    'package.json'
+    'package.json',
+    '**/msnodesqlv8/build/Release/*.node'
```
Comment (not-night-but) [inline apps/studio/electron-builder-config.js:41]: I'm not sure this is necessary?

**Review by not-night-but — CHANGES_REQUESTED** [review]
@mattpcanales Okay so I think the code mostly looks good, but I can't actually seem to connect with windows auth. It just hangs indefinitely, so we should probably make sure there's at least a connection timeout there (also if you know of anything that can cause that kind of issue please let me know. We should upgrade the `mssql` package so we can actually use the configuration object for the drivers. And I don't love how many changes there are to the `yarn.lock` It should not be that big of a diff. It looks like something has caused issues for yarn installs on other platforms

**Comment by rathboma** [comment]
Closing this as support shipped with Beekeeper Studio v6 for both windows auth and kerberos
